### PR TITLE
PUB-1773 Bulk Delete Subscription Functional Tests

### PR DESCRIPTION
**Pull request by junaidiqbalmoj**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-1773

### Change description ###

Add functional test for bulk subscription delete functionality.

**Does this PR introduce a breaking change?** (check one with "x")
[ ] Yes
[x] No

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionControllerTests.java [hunk 1]
```diff
@@ -727,6 +728,65 @@ void testBuildDeletedArtefactSubscribersReturnsAccepted() throws Exception {
         );
     }
 
+    @Test
+    void testBuildBulkDeletedSubscribersReturnsAccepted() throws Exception {
```
Comment (ChrisS1512) [inline src/integrationTest/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionControllerTests.java:732]: We should also test that the subscription has been deleted here by calling the get endpoint to make sure it's gone

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionControllerTests.java (hunk 1 above)
Comment (junaidiqbalmoj) [reply src/integrationTest/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionControllerTests.java:732]: done

**Review by ChrisS1512 — APPROVED** [review]

**Review by furnivall — DISMISSED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionControllerTests.java [hunk 2]
```diff
@@ -727,6 +728,85 @@ void testBuildDeletedArtefactSubscribersReturnsAccepted() throws Exception {
...
+        assertEquals(HttpStatus.NOT_FOUND.value(),
+                     getCaseSubscriptionResponse.getResponse().getStatus(),
+                     FORBIDDEN_STATUS_CODE
+        );
+        MvcResult getLocationSubscriptionResponse =
+            mvc.perform(getSubscriptionByUuid(locationSubscriptionId))
+            .andExpect(status().isNotFound()).andReturn();
+        assertEquals(HttpStatus.NOT_FOUND.value(),
+                     getLocationSubscriptionResponse.getResponse().getStatus(),
+                     FORBIDDEN_STATUS_CODE
```
Comment (ashwini-mv) [inline src/integrationTest/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionControllerTests.java:767]: For 'not found' status code we can not throw FORBIDDEN_STATUS_CODE message. Either it should be specific to not found error message or more generic one like "Status codes do not match"

File: src/integrationTest/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionControllerTests.java [hunk 3]
```diff
@@ -727,6 +728,85 @@ void testBuildDeletedArtefactSubscribersReturnsAccepted() throws Exception {
...
+        MvcResult response = mvc.perform(delete(DELETED_BULK_SUBSCRIPTION_PATH)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(subscriptionIdRequest))
+            .andExpect(status().isNotFound()).andReturn();
+
+        assertEquals(HttpStatus.NOT_FOUND.value(), response.getResponse().getStatus(),
+                     FORBIDDEN_STATUS_CODE
+        );
+    }
+
```
Comment (ashwini-mv) [inline src/integrationTest/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionControllerTests.java:785]: Same as above

File: src/integrationTest/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionControllerTests.java (hunk 2 above)
Comment (junaidiqbalmoj) [reply src/integrationTest/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionControllerTests.java]: fixed. change to "Status code does not match not found"

File: src/integrationTest/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionControllerTests.java (hunk 3 above)
Comment (junaidiqbalmoj) [reply src/integrationTest/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionControllerTests.java:786]: fixed.

**Review by ChrisS1512 — APPROVED** [review]

**Review by ashwini-mv — APPROVED** [review]

**Review by ChrisS1512 — APPROVED** [review]

**Review by ashwini-mv — APPROVED** [review]

**Pull request merged**